src/hooks/useDelegatesInfo.ts

```diff
@@ -0,0 +1,107 @@
+import useAsyncMemo from 'decentraland-gatsby/dist/hooks/useAsyncMemo'
+import fetch from 'isomorphic-fetch'
+import { useMemo } from 'react'
+import { SNAPSHOT_QUERY_ENDPOINT, SNAPSHOT_SPACE } from '../entities/Snapshot/constants'
+import useVotingPowerBalance from './useVotingPowerBalance'
+
+const QUERY = `
+query ($space: String!, $address: [Bytes!]) {
+  delegatedFrom: delegations(
+    where: {space_in: ["", $space], delegate_in: $address}
+    orderBy: delegate
+    orderDirection: desc
+  ) {
+    delegator
+    delegate
+    space
+    timestamp
+  }
+}
+`
+type DelegationQueryResult = {
+  delegator: string
+  delegate: string
+  space: string
+}
+
+type PickedByResult = {
+  address: string
+  pickedBy: number
+}
+
+export type Delegate = {
+  address: string
+  pickedBy: number
+  totalVP: number
+}
+
+function getTotalVP(addresses: string[]) {
+  return addresses.map(addr => {
+    const [votingPower] = useVotingPowerBalance(addr, SNAPSHOT_SPACE)
+    return { [addr]: votingPower }
+  }).reduce((obj, item) => Object.assign(obj, item), {})
+}
+
+function getPickedBy(addresses: string[]) {
+  const initialValue: PickedByResult[] = []
+  return useAsyncMemo(async () => {
+    if (addresses.length === 0 || SNAPSHOT_SPACE === '') {
+      return initialValue
+    }
+
+    const request = await fetch(
+      SNAPSHOT_QUERY_ENDPOINT,
+      {
+        method: 'post',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
+          query: QUERY,
+          variables: { address: addresses, space: SNAPSHOT_SPACE },
+        }),
+      }
+    )
+
+    const body = await request.json()
```

Comment:
I'm wondering, what happens if the fetch fails? 🤔

Reply:
try-catch implemented!